### DISCUSSION TITLE
How many USB interfaces are possible

### DISCUSSION THREAD

**Discussion by ccrome** [sekigon-gonnoc/Pico-PIO-USB#72] [General]
Has anyone attempted to have more than one PIO based USB host/device?  

The readme states, "Two PIO" but does that mean "Two PIO State Machines" or "Two PIO blocks"?  If the former, is it possible to get fully 4 bitbang'd usb busses going?  That would be pretty sweet.

**Comment by ccrome** [comment]
Ah, I think i've answered my own question:  two PIOs are required, and there are only two PIOs on the RP2040, so limited to one extra USB device, correct?

Anyway, it's pretty amazing that this works!  I just brought it up, and I can even attach multiple mice via a hub!  Very cool!

**Reply by ghost** [reply]
Hi @ccrome 
there is a function called `pio_usb_host_add_port` which adds a new host port. This can even be extendet to more ports if the value `PIO_USB_ROOT_PORT_CNT` (in file `pio_usb_configuration.h`) is changed (from 2 to 4 for example).
`pio_usb_host_add_port` needs only 2 parameters (PIO_USB2_DP_PIN, PIO_USB_PINOUT_DPDM) and should be called after `  tuh_init(BOARD_TUH_RHPORT)`.
By the way, this has got nothing to do with further PIOs and state machines. Its internally done by changing pin settings on the fly (jmp pin, in pin, set pins, sideset pins).
But this works only for host ports (for now).

**Comment by AbhishekAKA47** [comment]
can you help me in connecting a mouse to the pico and pico to the computer so its basically does nothing as of now. i am a novice
